packages/Sandblocks-VisualPrimitives/SBVisualPrimitive.class.st

```diff
@@ -609,9 +621,9 @@ SBVisualPrimitive >> propertyAt: aSymbol [
 	(self labelPositions includes: aSymbol) ifTrue: [
 		^ self blockDict
 			at: aSymbol
-			ifPresent: [:block | self getLabelContents: block]
+			ifPresent: [:block | (self vpMappingAt: aSymbol) getLabelContents: block]
 			ifAbsent: [nil]].
-	aSymbol = #children ifTrue: [^ self children].
+	aSymbol = #children ifTrue: [^ self children collect: #buildObject].
```

Comment:
two things here:
* for one, we need to have the mapping decide how to return values (closure need to produce source code and parse it for example)
* also, the children were returned as VPs instead of the underlying sourceObjects. Not sure if changing it here is the right call?